### PR TITLE
Merge zarhus layers

### PR DESCRIPTION
**Pull request by macpijan**
Proposal of merging back the layers into a single repository. With 3 moderately active repositories we already have some usage and maintenance issues. Having all of the public zarhus layers under one repo should be easier to maintain in the long run.

### CODE REVIEW

**Comment by DaniilKl** [comment]
So, you propose to merge BSP layers as well?

**Comment by macpijan** [comment]
> So, you propose to merge BSP layers as well?

I think so, at least it becomes really problematic practically.

Our goal would be to use `meta-zarhus` for all projects.

W can carry all layers in a single repo, but we do not have to enable them for all build, as enabling layers is done in kas files.

File: meta-zarhus-bsp-rockchip/README.md
Comment (tym2k1) [inline meta-zarhus-bsp-rockchip/README.md]: I think we should create a `meta-zarhus-bsp` layer instead of creating a separate layer for every SoC. Take a look at how poky handles it in [meta-yocto-bs](https://github.com/yoctoproject/poky/tree/master/meta-yocto-bsp). For now I don't see any drawbacks to their approach.

File: kas/common.yml
Comment (tym2k1) [inline kas/common.yml:10]: There should be a info in the main README.md on how to build images with the news kas files. I'm also in favour of keeping the build modular and [passing multiple `yml` files at once](https://kas.readthedocs.io/en/latest/userguide/project-configuration.html#including-configuration-files-via-the-command-line).

**Review by tym2k1 — CHANGES_REQUESTED** [review]
For now I'm leaving this as the 1st review. I still need to get through the merged layers and see if they're the same as mentioned rev's of the repositories from which they came from.

File: meta-zarhus-bsp-rockchip/README.md
Comment (DaniilKl) [reply meta-zarhus-bsp-rockchip/README.md]: IMHO, only scalability hods us from doing this.

File: kas/common.yml
Comment (tym2k1) [reply kas/common.yml:10]: This could also be combined with not having a specific yaml for targets but pass them through `--target` flag as described in https://github.com/zarhus/zarhus-issues/issues/10#issuecomment-2270813452

**Review by tym2k1 — COMMENTED** [review]
The [.pre-commit-config.yml](https://github.com/zarhus/meta-zarhus-bsp-rockchip/blob/main/.pre-commit-config.yaml) and [.oelint-ruleset.json](https://github.com/zarhus/meta-zarhus-bsp-rockchip/blob/main/.oelint-ruleset.json) from [meta-zarhus-bsp-rockchip](https://github.com/zarhus/meta-zarhus-bsp-rockchip/tree/main) differ slightly.

`.oelint-ruleset.json`
```diff
--- meta-zarhus-bsp-rockchip/.oelint-ruleset.json 2024-08-13 12:35:22.175008588 +0200
+++ meta-zarhus/.oelint-ruleset.json     2024-08-12 15:05:25.419067577 +0200
@@ -134,7 +134,7 @@
   "oelint.vars.fileextrapathsop": "error",
   "oelint.var.filesoverride": "warning",
   "oelint.vars.homepageprefix": "warning",
-  "oelint.vars.homepageping": "warning",
+  "oelint.vars.homepageping": "info",
   "oelint.vars.insaneskip": "info",
   "oelint.var.licenseremotefile": "info",
   "oelint.vars.licensesdpx": "warning",
```

`.pre-commit-config.yml`
```diff
--- meta-zarhus-bsp-rockchip/.pre-commit-config.yaml  2024-08-13 12:35:22.175008588 +0200
+++ meta-zarhus/.pre-commit-config.yaml  2024-08-12 15:05:25.419067577 +0200
@@ -6,7 +6,7 @@

 repos:
   - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v4.4.0
+    rev: v4.6.0
     hooks:
       - id: check-added-large-files
       - id: check-merge-conflict
@@ -17,19 +17,19 @@
       - id: mixed-line-ending

   - repo: https://github.com/adrienverge/yamllint
-    rev: v1.32.0
+    rev: v1.35.1
     hooks:
       - id: yamllint
         args: [-c=.yamllint]

   - repo: https://github.com/igorshubovych/markdownlint-cli
-    rev: v0.36.0
+    rev: v0.41.0
     hooks:
       - id: markdownlint
       - id: markdownlint-fix

   - repo: https://github.com/talos-systems/conform
-    rev: v0.1.0-alpha.27
+    rev: v0.1.0-alpha.30
     hooks:
       - id: conform
         stages:
@@ -41,8 +41,8 @@
       - id: shellcheck
         args: ["--severity=warning"]

-  - repo: https://github.com/zarhus/oelint-adv
-    rev: 5.5.2-zarhus
+  - repo: https://github.com/priv-kweihmann/oelint-adv
+    rev: 5.7.2
     hooks:
       - id: oelint-adv
         args: [--rulefile=.oelint-ruleset.json, --hide=info, --quiet, --fix]
@@ -54,7 +54,7 @@
         files: .*\.(bb)|(bbappend)|(bbclass)$

   - repo: https://github.com/codespell-project/codespell
-    rev: v2.2.5
+    rev: v2.3.0
     hooks:
       - id: codespell

@@ -63,6 +63,7 @@
     hooks:
       - id: namespell
         args: [--fix]
+        exclude: README.md

 ci:
   autoupdate_commit_msg: 'pre-commit: autoupdate hooks'
```

Which version should we use?

**Comment by DaniilKl** [comment]
@tym2k1, IMHO with the newest versions with `README.md` excluded. You should also make sure `oelint.vars.homepageping` does not produce any warnings.

File: meta-zarhus-bsp-rockchip/README.md
Comment (tym2k1) [reply meta-zarhus-bsp-rockchip/README.md]: It was decided that separate platforms will need their own layers so we'll store them in a [meta-zarhus-bsp](https://github.com/zarhus/meta-zarhus/pull/18/commits/9beb6fd78fefad886cadec1e89e6f536c2d85698) folder

File: kas/common.yml
Comment (tym2k1) [reply kas/common.yml:10]: For now handled with separate `.yml` files. The `--target` parts is TBD. Related commit - https://github.com/zarhus/meta-zarhus/pull/18/commits/67c2b22fe583981582124bb9a7284d83c7105cef

**Review by tym2k1 — APPROVED** [review]

**Comment by DaniilKl** [comment]
@tym2k1, have the commits you have added gone through review of another reviewer before merging?

**Comment by tym2k1** [comment]
@DaniilKl There was an internal talk about it and I merged it to the newly created `develop` branch not `main`. If you have any concerns please let's continue the discussion in an issue.

**Comment by DaniilKl** [comment]
I am not talking about concepts we have talked about, but about not merging **anything** without peer review.

**Comment by tym2k1** [comment]
Please let's discuss this matter outside of this thread. I might've decided too rashly to merge this PR but I still talked to both @macpijan and you about the changes in https://github.com/zarhus/meta-zarhus/pull/21 and adressed all threads. I've also self-signed as a reviewer after an internal talk not waiting for someone to tag me as a reviewer. I'm afraid that too much hand-holding might be detrimental to the workflow of the project but it's a matter of broader discussion.

**Pull request merged**
